/*
 * Copyright (c) 2016 Cisco and/or its affiliates, and
 *                    Cable Television Laboratories, Inc. ("CableLabs")
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at:
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

 /*
  * This file is a copy of rcp/rcp_lib/CcapCoreIdentification.proto
  */

syntax = "proto2";


    message t_CcapCoreIdentification {
        required uint32 Index = 2;
        optional bytes CoreId = 3;
        optional string CoreIpAddress = 4;
        optional bool IsPrincipal = 5;
        optional string CoreName = 6;
        optional uint32 VendorId = 7;

        enum t_CoreMode {
            COREMODEACTIVE = 1;
            COREMODEBACKUP = 2;
            COREMODENOTACTING = 3;
            COREMODEDECISIONPENDING = 4;
            COREMODEOUTOFSERVICE = 5;
            COREMODECONTACTPENDING = 6;
        }

        optional t_CoreMode CoreMode = 8;
        optional bool InitialConfigurationComplete = 9;
        optional bool MoveToOperational = 10;
        optional uint32 CoreFunction = 11;
        optional uint32 ResourceSetIndex = 12;
    }
